### ISSUE TITLE
Uncaught exception - Invalid call (MPHX broken on Haxe 3.3)

### ISSUE THREAD

**Issue by ediblebird** [5Mixer/mphx#34]
Hi! I am trying to run the basic server example, and when i do i get this error

> Server.hx:76: Server active on 127.0.0.1:8000. Code after server.start() will not run.
> Called from ? line 1
> Called from Main.hx line 47
> Called from Main.hx line 7
> Called from Main.hx line 38
> Called from mphx/server/impl/Server.hx line 80
> Called from mphx/server/impl/Server.hx line 99
> Called from C:\dev\HaxeToolkit\haxe\std/neko/_std/sys/net/Socket.hx line 241
> Called from C:\dev\HaxeToolkit\haxe\std/neko/_std/sys/net/Socket.hx line 235
> Uncaught exception - Invalid call

I am using the git version of mphx
haxe 3.30
compiled to neko on Windows 7 64bit

any help would be appreciated!

**Comment by 5Mixer** [comment]
Thanks for raising the issue @ediblebird, I'm glad I got told about this.

So I ran the sample on version 3.2 of Haxe, which was the version I have been developing on, and it worked fine. Updating to 3.3 and running the sample through the error you are seeing. So obviously the issue was introduced with haxe changes. 

Haxe offers a guide for finding the breaking haxe commit [here](https://github.com/HaxeFoundation/haxe/wiki/Finding-breaking-commits-with-git-bisect) but I'd like to find the problem manually if I can. I'd like for this to be a super easy case of 'just update!' but it looks like I might need to put in a bit of work digging around the code base. The change might be super easy to catch, I'm not sure.

I'll try to get on to this as soon as I can. Thanks!

Edit: As a general note for other MPHX users, any changes updating to mphx for 3.3 may break mphx for 3.2 haxe, so be aware/update haxe.

**Comment by 5Mixer** [comment]
Alright, so as of now, this is in it's 'most solved' state possible. The issue was on the haxe end of things, with `socket.select` being broken on the neko target. An issue was raised there, and resolved with a commit to the haxe repository.

The 'issue' with this though, is `socket.select` will always be broken on haxe 3.3.0, and thus, mphx will be too (unless some unlikely neko workaround pops up). However, every version after 3.3.0 should be fixed. I've tested this by downloading the nightly build of haxe, which consists of the very latest haxe changes, and it works.

![ba130f345c1e79b2f2e28294df3dced0](https://cloud.githubusercontent.com/assets/8501694/17019712/0c4fdd68-4f81-11e6-9c1a-e37ca0da18a8.png)

You can try out the latest haxe version by downloading the nightly build from the [build site](http://hxbuilds.s3-website-us-east-1.amazonaws.com/builds/haxe/windows-installer/). (Unzip with something like 7zip, then run the installer in the folder).

For the sake of documentation, I'll upload how this was kindly explained by nakado in this chat ([1](https://i.gyazo.com/30fb9e2398b1e2b4c96c1e2dfdf96004.png) [2](https://i.gyazo.com/9a6d5851f3abf2a83c5cb81b0c2431c5.png))

Again, versioning means that I can't change the fact that 3.3.0 is broken, I could just fix the following updates.

Thanks for your patience @pxshadow, I've been super busy irl.

tl;dr: It's fixed! :smile:

**Comment by PXshadow** [comment]
THANK YOU so much.
